src/ztc/api/validators.py

```diff
@@ -194,3 +194,16 @@ def __call__(self, attrs):
                             f"Objects can't be updated with a relation to non-concept {field_name}"
                         )
                         raise ValidationError(msg, code=self.code)
+
+
+class ZaakInformatieObjectTypeCatalogusValidator:
+    code = "relations-incorrect-catalogus"
+    message = _("The zaaktype has catalogus different from informatieobjecttype")
+
+    def __call__(self, attrs: dict):
+        zaaktype = attrs.get("zaaktype")
+        informatieobjecttype = attrs.get("informatieobjecttype")
```

Comment:
Something that I have missed in my own PR for this in openzaak as well (because there were no PUT and PATCH possible there yet), but I think it would be better to take the zaak/informatieobjecttype from the attributes, and if it is not present in the attributes (i.e. a PATCH is done with just `zaaktype` and not `informatieobjecttype`), take it from the instance itself (i.e. `zaaktype = attrs.get("zaaktype") or instance.zaaktype`).

A test for this behaviour with PATCH would also be needed I think, I will add these tests for this validator in open-zaak as well

Reply:
Looks good, only thing I'm missing is the same test you added but then with a patch on the ZIOT with `zaaktype`, then I'll shut up :p